client/Components/eggDetails.jsx

```diff
@@ -46,8 +46,8 @@ const EggDetails = (props) => {
           createdAt: props.targetEgg.createdAt,
         };
         user.data.foundEggs.push(foundEgg);
-        setIsClaimed(true);
       })
+      .then(setIsClaimed(true))
```

Comment:
This is happening before the fetch completes. .then() takes a function definition as its argument. here you are passing the return of a function call.